backupccl,cloud: integrate s3 uploader, file sst sink with admission control

irfansharif opened the pull request:
**cloud/amazon: integrate s3 uploader with admission control**

We've seen the underlying upload process consume excess CPU during
backups (computing checksums, etc.), affecting scheduling latencies.

**backupccl: integrate the file sst sink with admission control**

It's a tight loop, and we've seen it show up prominently in CPU profiles
during backups.

**admission,schedulerlatency: allow configuration of ...**

.. scheduling latency percentile target for the elastic CPU limiter.
We've seen workloads sensitive to p99.9 go scheduler latency, further in
the tail than the p99 we've been using so far. We add a
scheduler_latency.sample_percentile to make this configurable.

Release note: None.

irfansharif committed:
admission,schedulerlatency: allow configuration of ...

.. scheduling latency percentile target for the elastic CPU limiter.
We've seen workloads sensitive to p99.9 go scheduler latency, further in
the tail than the p99 we've been using so far. We add a
scheduler_latency.sample_percentile to make this configurable.

Release note: None

irfansharif committed:
backupccl: integrate the file sst sink with admission control

It's a tight loop, and we've seen it show up prominently in CPU profiles
during backups.

Release note: None

irfansharif committed:
cloud/amazon: integrate s3 uploader with admission control

We've seen the underlying upload process consume excess CPU during
backups (computing checksums, etc.), affecting scheduling latencies.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/108231)
<!-- Reviewable:end -->

irfansharif committed:
roachtest: admission-control/elastic-backup with s3

dt committed:
backup: add env var to discard data

The env var COCKROACH_BACKUP_TESTING_DISCARD_DATA can be set to tell the backup process to discard all data
during a backup, rather than writing it to the backup destination. Metadata is still tracked and written,
but the row data SST files are discarded.

Release note: none.
Epic: none.

irfansharif committed:
[wip] cloud: lower ToS for upload traffic

irfansharif committed:
[wip] cloud: further lower ToS for upload traffic
